### PR TITLE
Feature/patch fixes

### PR DESCRIPTION
**Pull request by mespinoza510**
Description: Made changes miscellaneous changes regarding your comments @jantimar. Still want to know how I can change the `screenWidth` and `screenHeight` using `GeometryReader`

### CODE REVIEW

File: Native-Frameworks/Screens/DetailView/FrameworkDetailView.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ struct FrameworkDetailView: View {
             } label: {
                 FrameworksButton(title: "More Info")
             }
-        }.fullScreenCover(isPresented: $isShowingSafariView, content: {
+        }.sheet(isPresented: $isShowingSafariView, content: { // TODO: You show screen `fullScreenCover` under `fullScreenCover`, I think you should call only 1 time, probably in first place in your Grid you should use list and push your screen
```
Comment (mespinoza510) [inline Native-Frameworks/Screens/DetailView/FrameworkDetailView.swift]: @jantimar I changed to `sheet` to not call `fullScreenCover` twice. I want to later implement changing from `GridView` to `ListView` with my `popup`.

File: Native-Frameworks/FrameworkGridView.swift [hunk 2]
```diff
@@ -23,18 +23,22 @@ struct FrameworkGridView: View {
...
+                            isShowingDetailView: $viewModel.isShowingDetailView,
+                            framework: viewModel.selectedFramework!
+                        )
                     }
                 }
                 
             }
             //TODO: Refactor logic to viewModel
+            // Note: Its view and it don't and I dont see any extra logic, but probably you can move it to separete property
             if isPressed {
```
Comment (mespinoza510) [inline Native-Frameworks/FrameworkGridView.swift:68]: @jantimar I was a bit confused on what you meant by storing it in separate property? Any suggestions?

File: Native-Frameworks/FrameworkGridView.swift [hunk 3]
```diff
@@ -23,18 +23,22 @@ struct FrameworkGridView: View {
                     .navigationTitle("🍎 Frameworks")
                     .navigationBarItems(trailing: EllipsisButton(isPressed: $isPressed))
                     .fullScreenCover(isPresented: $viewModel.isShowingDetailView) {
-                        FrameworkDetailView(isShowingDetailView: $viewModel.isShowingDetailView, framework: viewModel.selectedFramework ?? MockData.sampleFramework)
+                        FrameworkDetailView(
+                            isShowingDetailView: $viewModel.isShowingDetailView,
+                            framework: viewModel.selectedFramework!
```
Comment (jantimar) [inline Native-Frameworks/FrameworkGridView.swift]: in `FrameworkGridViewModel` implement  `selectedFramework`  default value so it should never be nil / optional

File: Native-Frameworks/Screens/FrameWorkGridViewModel.swift [hunk 4]
```diff
@@ -1,14 +1,15 @@
 import SwiftUI
 
 final class FrameworkGridViewModel: ObservableObject {
-    
-    var selectedFramework: Framework? {
+
+    // TODO: I think `selectedFramework` and `isShowingDetailView` can be join to one property `Freamwork` of `@Published` type and check if this value is nil ( I am not sure it will be work on 100% )
+    var selectedFramework: Framework? = MockData.sampleFramework {
```
Comment (jantimar) [inline Native-Frameworks/Screens/FrameWorkGridViewModel.swift]: `Framework?` -> `Framework`

File: Native-Frameworks/Utilities/SystemImageName.swift [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+public enum SystemImageName: String {
+    case ellipsis = "ellipsis"
```
Comment (jantimar) [inline Native-Frameworks/Utilities/SystemImageName.swift]: ` case ellipsis = "ellipsis"` -> ` case ellipsis`

File: Native-Frameworks/Utilities/SystemImageName.swift [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+public enum SystemImageName: String {
```
Comment (jantimar) [inline Native-Frameworks/Utilities/SystemImageName.swift]: Why is this `public` ?

File: Native-Frameworks/Views/CustomColors/BackgroundColor.swift [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import SwiftUI
+
+struct BackgroundColor: View {
+    var body: some View {
+        // Setup display for custom color
+        ZStack {
+            Rectangle()
+                .frame(width: 300, height: 300)
```
Comment (jantimar) [inline Native-Frameworks/Views/CustomColors/BackgroundColor.swift]: You call more times frame with same width and height probalby this extension can help you 

``` 
extension View {

 func frame(side: CGFloat) -> some View {
     return self.frame(width: side, height: side)
  }
}
```

than you can set frame on rectancgle `Rectangle().frame(side: 300)`

File: Native-Frameworks/Views/CustomColors/BackgroundColor.swift [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+struct BackgroundColor_Previews: PreviewProvider {
+    static var previews: some View {
+        BackgroundColor()
+    }
+}
+
+extension Color {
+    public static var lightBlack: Color {
```
Comment (jantimar) [inline Native-Frameworks/Views/CustomColors/BackgroundColor.swift]: Why public?

Comment (mespinoza510) [reply Native-Frameworks/Views/CustomColors/BackgroundColor.swift]: Removed Public

File: Native-Frameworks/Views/CustomColors/BackgroundColor.swift (hunk 7 above)
Comment (mespinoza510) [reply Native-Frameworks/Views/CustomColors/BackgroundColor.swift]: called the method `squareFrame` to fit the description correctly.

File: Native-Frameworks/FrameworkGridView.swift (hunk 3 above)
Comment (mespinoza510) [reply Native-Frameworks/FrameworkGridView.swift]: Removed optional and the force unwrapping of `selectedFramework`

File: Native-Frameworks/Views/Buttons/XDismissButton.swift [hunk 9]
```diff
@@ -4,14 +4,16 @@ struct XDismissButton: View {
...
         HStack {
             Spacer()
             
             Button {
                 isShowingModal = false
             } label: {
-                Image(systemName: "xmark")
+                Image(systemName: xMarkImage)
                     .foregroundColor(Color(.label))
                     .frame(width: 45, height: 45)
```
Comment (jantimar) [inline Native-Frameworks/Views/Buttons/XDismissButton.swift]: please use `squareFrame`

File: Native-Frameworks/Views/Buttons/EllipsisButton.swift [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    var body: some View {
+        Button {
+            withAnimation {
+                self.isPressed.toggle()
+            }
+        } label: {
+            Image(systemName: ellipsisImage)
+                .resizable()
+                .scaledToFit()
+                .frame(width: 30, height: 30)
```
Comment (jantimar) [inline Native-Frameworks/Views/Buttons/EllipsisButton.swift]: Use `squareFrame`

File: Native-Frameworks/FrameworkGridView.swift [hunk 11]
```diff
@@ -23,18 +23,22 @@ struct FrameworkGridView: View {
...
+                            isShowingDetailView: $viewModel.isShowingDetailView,
+                            framework: viewModel.selectedFramework
+                        )
                     }
                 }
                 
             }
             //TODO: Refactor logic to viewModel
+            // Note: Its view and it don't and I dont see any extra logic, but probably you can move it to separete property
             if isPressed {
```
Comment (jantimar) [inline Native-Frameworks/FrameworkGridView.swift:68]: You add new variable (or func) in this sturc f.e.

```
private var selectionView: some View {
  GeometryReader { geo in
  ...
}
```

and then on line 36
```
if isPressed {
  selectionView
}
```

File: Native-Frameworks/FrameworkGridView.swift [hunk 12]
```diff
@@ -53,29 +57,8 @@ struct FrameworkGridView_Previews: PreviewProvider {
...
-                .rotationEffect(.degrees(90))
-                .foregroundColor(Color(.label))
-        }
-    }
-}
-
+// TODO: Can you replace this constants with `GeometryReader`
 extension UIScreen {
     static let screenWidth = UIScreen.main.bounds.size.width
     static let screenHeight = UIScreen.main.bounds.size.height
```
Comment (jantimar) [inline Native-Frameworks/FrameworkGridView.swift]: 1) Prefer georeader
2) Please look on `typealias` this will be better as static let property 
`typealias screenWidth = UIScreen.main.bounds.size.width`

File: Native-Frameworks/FrameworkGridView.swift [hunk 13]
```diff
@@ -23,18 +23,22 @@ struct FrameworkGridView: View {
...
                 }
                 
             }
             //TODO: Refactor logic to viewModel
+            // Note: Its view and it don't and I dont see any extra logic, but probably you can move it to separete property
             if isPressed {
                 GeometryReader { geo in
                     PopupSelectionView(isGridviewSelected: $isGridviewSelected, isListviewSelected: $isListviewSelected)
                 }
                 .offset(x: UIScreen.screenWidth * 0.61, y: UIScreen.screenHeight * 0.1)
```
Comment (jantimar) [inline Native-Frameworks/FrameworkGridView.swift]:
```
GeometryReader { reader in
 PopupSelectionView(isGridviewSelected: $isGridviewSelected, isListviewSelected: $isListviewSelected)		                  
                  .offset(x: reader.size.width * 0.61, y: reader.size.height * 0.1)
}
```
In the current case your reader sizes will be size of your popup, so the width should work, I hope,  If you want the size of FrameworkGridView you should wrap all your content in GeometryReader

Comment (mespinoza510) [reply Native-Frameworks/FrameworkGridView.swift]: If you check latest commit and using this solution - it works. Let me know when it looks good to merge

File: Native-Frameworks/FrameworkGridView.swift (hunk 12 above)
Comment (mespinoza510) [reply Native-Frameworks/FrameworkGridView.swift]: used georeader as per my latest commits.

File: Native-Frameworks/FrameworkGridView.swift (hunk 13 above)
Comment (jantimar) [reply Native-Frameworks/FrameworkGridView.swift]: If it works on all devices I accept it :) please try it on SE 12 and 12 Pro

**Review by jantimar — APPROVED** [review]

File: Native-Frameworks/FrameworkGridView.swift [hunk 14]
```diff
@@ -6,12 +6,28 @@ struct FrameworkGridView: View {
     @State private var isPressed = false
     @State private var isGridviewSelected = true
     @State private var isListviewSelected = false
+    @State private var isPopupDisplayed = false
```
Comment (jantimar) [inline Native-Frameworks/FrameworkGridView.swift:9]: Why did you add this property? You never use it

File: Native-Frameworks/FrameworkGridView.swift [hunk 15]
```diff
@@ -6,12 +6,28 @@ struct FrameworkGridView: View {
...
+            Spacer()
+            
+            VStack {
+                PopupSelectionView(isGridviewSelected: $isGridviewSelected, isListviewSelected: $isListviewSelected)
+                
+                Spacer()
+            }
+            .padding(.top, 100)
         }
+        .frame(maxWidth: UIScreen.screenWidth, maxHeight: UIScreen.screenHeight)
```
Comment (jantimar) [inline Native-Frameworks/FrameworkGridView.swift:23]: I don't accept this change, prefer infinity to be full screen, not you still using UIKit and we do it to remove UIKit from this project
